src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -537,10 +560,84 @@ public Void run(GitClient client, String remoteName) throws IOException, Interru
                     if (context.wantTags()) {
                         discoverTags(repository, walk, request, remoteReferences);
                     }
+                    if (context.wantOtherRefs()) {
+                        discoverOtherRefs(repository, walk, request, remoteReferences,
+                                (Collection<GitSCMSourceContext.WantedOtherRef>)context.getOtherWantedRefs());
```

Comment:
`WantedOtherRef` is not a name I like, especially as it leaks into the public API... any suggested alternatives? (if it is the least bad, well we'll have to live with it)

Reply:
`AdditionalRefSpec` or something like that might be better... or worse... suggestions please... and then we pick the least ugly

Reply:
`RefNameMapping` aligns better with the usage that gets logged

Reply:
`AdditionalRefSpec` is already used in a method somewhere.
The use of `wanted` was to reflect that it is about the same as `wantBranches` and `wantTags`. And I guess my brain got stuck on that train :)

`RefNameMapping` sounds better, I'll chew on it for a bit.